src/frontend/src/components/LinkEditView.tsx

```diff
@@ -0,0 +1,69 @@
+import { useAllLinkTypes } from '../hooks/projects.hooks';
+import LoadingIndicator from './LoadingIndicator';
+import ErrorPage from '../pages/ErrorPage';
+import { Button, Grid, IconButton, MenuItem, TextField } from '@mui/material';
+import { FieldArrayWithId, UseFieldArrayAppend, UseFieldArrayRemove, UseFormRegister } from 'react-hook-form';
+import DeleteIcon from '@mui/icons-material/Delete';
+import { LinkCreateArgs } from 'shared';
+
+const LinkEditView: React.FC<{
+  ls: FieldArrayWithId[];
+  register: UseFormRegister<{
+    name: string;
+    budget: number;
+    summary: string;
+    projectLeadId: number | undefined;
+    projectManagerId: number | undefined;
+    crId: string;
+    rules: { rule: string }[];
+    links: LinkCreateArgs[];
+    goals: { bulletId: number; detail: string }[];
+    features: { bulletId: number; detail: string }[];
+    constraints: { bulletId: number; detail: string }[];
+  }>;
+  append: UseFieldArrayAppend<any, any>;
+  remove: UseFieldArrayRemove;
+  links: LinkCreateArgs[];
+}> = ({ ls, register, append, remove, links }) => {
+  const { isLoading, isError, error, data } = useAllLinkTypes();
+  if (isLoading || !data) return <LoadingIndicator />;
+  if (isError) return <ErrorPage message={error.message} />;
+
+  return (
+    <>
+      {ls.map((_element, i) => {
+        return (
+          <Grid item sx={{ display: 'flex', alignItems: 'center', mb: '5px' }}>
+            <TextField
```

Comment:
@RChandler234 havnt implemented the rigid structure for editing the project with the links, specifically what are you thinking of.

Reply:
do you think it shouldnt be a use field array

Reply:
I'm fine with the use field array stuff, but there should be some links that are required on all projects (and we should have a corresponding tool in admin tools for admins to set this). I like how it looks in your screenshots, but we would just disable the ability to edit the type or delete for the required links (grey out the select with it set to the proper type, and remove the delete button)

Reply:
and then they would be able to add as many extra links as they want if they feel it's useful